targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52840/TARGET_EP_AGORA/ONBOARD_TELIT_ME910.h

```diff
@@ -31,6 +31,17 @@ class ONBOARD_TELIT_ME910 : public TELIT_ME910 {
     virtual nsapi_error_t hard_power_off();
     virtual nsapi_error_t soft_power_on();
     virtual nsapi_error_t soft_power_off();
+
+private:
+    void press_power_button(int time_ms);
+
+    void onboard_modem_init();
+
+    void onboard_modem_deinit();
+
+    void onboard_modem_power_up();
+
+    void onboard_modem_power_down();
```

Comment:
Should we consider making these functions public? Making them private restricts power optimization efforts at the application level which were previously available in the `onboard_modem_api.c` functions (e.g. turning off the cell module when not in use).

Reply:
I would expect the power control from application end to be via higher-level calls to the cellular framework - those calls were always viewed as "driver core to hardware glue" type calls, and would be only part of the power saving. (You'd want to do serial port suspension too).

Whether higher-level application appropriate calls currently exist is another matter.